tempodb/encoding/backend_block.go

```diff
@@ -125,3 +126,50 @@ func (b *BackendBlock) NewIndexReader() (common.IndexReader, error) {
 
 	return reader, nil
 }
+
+func (b *BackendBlock) BlockMeta() *backend.BlockMeta {
+	return b.meta
+}
+
+func (b *BackendBlock) Reader() backend.Reader {
+	return b.reader
+}
+
+func (b *BackendBlock) Write(ctx context.Context, r backend.ReaderReader, w backend.Writer) error {
```

Comment:
this method is copying from `ReaderReader` to `Writer` and uses the block meta and tenant id?

i think this will be confusing given that the "real" backend write methods are all in `block.go` and this is more like a copy from backend reader to writer method.

Reply:
Agree, moved to a general purpose method in `block.go`